editor/src/components/navigator/left-pane/pages-pane.tsx

```diff
@@ -0,0 +1,236 @@
+/** @jsxRuntime classic */
+/** @jsx jsx */
+import { jsx } from '@emotion/react'
+
+import { useAtom } from 'jotai'
+import React from 'react'
+import { matchRoutes } from 'react-router'
+import { uniqBy } from '../../../core/shared/array-utils'
+import * as EP from '../../../core/shared/element-path'
+import { NO_OP } from '../../../core/shared/utils'
+import { FlexColumn, FlexRow, Icn, Subdued, UtopiaTheme, colorTheme } from '../../../uuiui'
+import { RemixIndexPathLabel } from '../../canvas/remix/remix-utils'
+import {
+  ActiveRemixSceneAtom,
+  RemixNavigationAtom,
+} from '../../canvas/remix/utopia-remix-root-component'
+import { Substores, useEditorState } from '../../editor/store/store-hook'
+import { ExpandableIndicator } from '../navigator-item/expandable-indicator'
+import { getFeaturedRoutesFromPackageJSON } from '../../../printer-parsers/html/external-resources-parser'
+import { defaultEither } from '../../../core/shared/either'
+
+type RouteMatch = {
+  path: string
+  resolvedPath: string
+  matchesRealRoute: boolean
+}
+
+export const PagesPane = React.memo((props) => {
+  const featuredRoutes = useEditorState(
+    Substores.projectContents,
+    (store) => {
+      return defaultEither([], getFeaturedRoutesFromPackageJSON(store.editor.projectContents))
+    },
+    'PagesPane featuredRoutes',
+  )
+  const registeredExampleRoutes = fillInGapsInRoute(featuredRoutes)
+
+  const remixRoutes: Array<RouteMatch> = useEditorState(
+    Substores.derived,
+    (store) => {
+      const result = uniqBy(
+        registeredExampleRoutes.flatMap((exampleRoute): Array<RouteMatch> => {
+          const matchResult = matchRoutes(store.derived.remixData?.routes ?? [], exampleRoute) ?? []
+
+          if (
+            matchResult[matchResult.length - 1]?.pathname !==
```

Comment:
can the index go to `-1`?

Reply:
yep if there's no match

Reply:
would it make sense then to use `safeIndex`?

Reply:
Actually I wanted to use Array.prototype.get(-1), but it needs bumping our TS config.

I'm forgot what safeIndex does. would it enforce the question mark?

Reply:
no, it would give you back an `| undefined` that you can handle instead of throwing because of the negative index

Reply:
<img width="192" alt="image" src="https://github.com/concrete-utopia/utopia/assets/2226774/95357d9d-9cc6-4ae9-8343-dc5edf3b67b9">
I thought arrays don't throw on negative index!